plugins/vendor/zoom/lib/model/kZoomEventHanlder.php

```diff
@@ -40,33 +40,245 @@ public function parseEvent()
+				}
+				else
+				{
+					$transcriptProcessor = new kZoomTranscriptProcessor($this->zoomConfiguration[kZoomClient::ZOOM_BASE_URL],
+					                                                    $zoomVendorIntegration->getJwtToken(),
+					                                                    $zoomVendorIntegration->getRefreshToken(),
+					                                                    null, null, $zoomVendorIntegration->getAccessToken());
+					$transcriptProcessor->handleRecordingTranscriptComplete($event);
+				}
 				break;
 		}
 	}
+	
+	protected static function getZoomDropFolderId(kZoomEvent $event, $zoomVendorIntegration)
+	{
+		$dropFolderType = ZoomDropFolderPlugin::getDropFolderTypeCoreValue(ZoomDropFolderType::ZOOM);
+		$dropFolders = DropFolderPeer::retrieveEnabledDropFoldersPerPartner($zoomVendorIntegration->getPartnerId(), $dropFolderType);
+		foreach ($dropFolders as $dropFolder)
+		{
+			if ($dropFolder->getZoomVendorIntegrationId() == $zoomVendorIntegration->getId())
+			{
+				return $dropFolder->getId();
+			}
+		}
+		return null;
+	}
+	
+	protected static function createZoomDropFolderFile(kZoomEvent $event, $dropFolderId, $partnerId)
+	{
+		/* @var kZoomRecording $recording */
+		$recording = $event->object;
+		
+		$dropFolderFilesMap = self::loadDropFolderFiles($dropFolderId);
+		
+		$kMeetingMetaData = self::allocateMeetingMetaData($recording, $event);
+		$recordingFilesOrdered = self::orderRecordingFiles($recording->recordingFiles);
+		foreach ($recordingFilesOrdered as $recordingFilesPerTimeSlot)
+		{
+			$firstDFFileOnTimeSlot = true;
+			$isParentEntry = false;
+			$parentEntry = null;
+			$wasParentEntryFound = false;
+			/* @var kZoomRecordingFile $recordingFile*/
+			foreach ($recordingFilesPerTimeSlot as $recordingFile)
+			{
+				$fileName = $kMeetingMetaData->getUuid() . '_' . $recordingFile->id . ZoomHelper::SUFFIX_ZOOM;
+				if(!array_key_exists($fileName, $dropFolderFilesMap))
+				{
+					if(!ZoomHelper::shouldHandleFileTypeEnum($recordingFile->recordingFileType))
+					{
+						continue;
+					}
+					$kRecordingFile = self::allocateZoomRecordingFile($recordingFile, $event);
+					$zoomDropFolderFile = self::allocateZoomDropFolderFile($dropFolderId, $partnerId, $fileName, $recordingFile->fileSize,
+					                                                      $kMeetingMetaData, $kRecordingFile);
+					if ($firstDFFileOnTimeSlot)
+					{
+						$firstDFFileOnTimeSlot = false;
+						$parentEntry = self::getEntryByReferenceId(zoomProcessor::ZOOM_PREFIX . $kMeetingMetaData->getUuid(), $partnerId);
+						if ($parentEntry)
+						{
+							$wasParentEntryFound = true;
+						}
+						else
+						{
+							$parentEntry = self::createEntry($recording->uuid, $partnerId);
+						}
+					}
+					
+					$zoomDropFolderFile->setParentEntryId($parentEntry->getId());
+					if (!$wasParentEntryFound && !$isParentEntry && $recordingFile->recordingFileType == kRecordingFileType::VIDEO)
```

Comment:
the logic of setting parent entry is not clear.
Why do we need parentEntryId and isParentEntry
 isParentEntry = (parentEntryId === id)